helper-cli/src/main/kotlin/commands/provenancestorage/DeleteCommand.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.helper.commands.provenancestorage
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.terminal
+import com.github.ajalt.clikt.parameters.options.associate
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+import com.github.ajalt.mordant.rendering.Theme
+import com.github.ajalt.mordant.terminal.YesNoPrompt
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.config.OrtConfiguration
+import org.ossreviewtoolkit.scanner.ScanStorages
+import org.ossreviewtoolkit.utils.common.expandTilde
+import org.ossreviewtoolkit.utils.ort.ORT_CONFIG_FILENAME
+import org.ossreviewtoolkit.utils.ort.ortConfigDirectory
+
+internal class DeleteCommand : CliktCommand(
+    help = "Deletes stored provenance results matching the options."
+) {
+    private val configFile by option(
+        "--config",
+        help = "The path to the ORT configuration file that configures the scan storages."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .default(ortConfigDirectory.resolve(ORT_CONFIG_FILENAME))
+
+    private val configArguments by option(
+        "-P",
+        help = "Override a key-value pair in the configuration file. For example: " +
+            "-P ort.scanner.storages.postgres.connection.schema=testSchema"
+    ).associate()
+
+    private val packageId by option(
+        "--package-id",
+        help = "Coordinates of the package ID to delete."
+    ).convert { Identifier(it) }
+        .required()
+
+    private val dryRun by option(
```

Comment:
I believe the dry-run is not really needed because of the yes/no prompt.
Maybe we could turn it into a `-y` or `-f` flag for saying yes to the prompt?